File: src/CodeCharacter.Core/Entities/PublicUserEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

#pragma warning disable CS8618

namespace CodeCharacter.Core.Entities;

public class PublicUserEntity
{
    public int AvatarId { get; set; }
    public string College { get; set; }
    public string Country { get; set; }
    public string Name { get; set; }

    [Key] [ForeignKey("User")] public int UserId { get; set; }
}

File: server/src/main/kotlin/delta/codecharacter/server/user/public_user/PublicUserEntity.kt
package delta.codecharacter.server.user.public_user

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "publicUser")
data class PublicUserEntity(
    @Id
    val username: String,
    val name: String,
    val country: String,
    val college: String,
    val avatarId: Int
)
